cmd/cloud_upload.go

```diff
@@ -0,0 +1,189 @@
+	)
+	printBar(c.gs, progressBar)
+
+	test, err := loadAndConfigureLocalTest(c.gs, cmd, args, getPartialConfig)
+	if err != nil {
+		return err
+	}
+
+	// It's important to NOT set the derived options back to the runner
+	// here, only the consolidated ones. Otherwise, if the script used
+	// an execution shortcut option (e.g. `iterations` or `duration`),
+	// we will have multiple conflicting execution options since the
+	// derivation will set `scenarios` as well.
+	testRunState, err := test.buildTestRunState(test.consolidatedConfig.Options)
+	if err != nil {
+		return err
+	}
+
+	// TODO: validate for usage of execution segment
+	// TODO: validate for externally controlled executor (i.e. executors that aren't distributable)
+	// TODO: move those validations to a separate function and reuse validateConfig()?
+
+	modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Building the archive..."))
+	arc := testRunState.Runner.MakeArchive()
+
+	tmpCloudConfig, err := cloudapi.GetTemporaryCloudConfig(arc.Options.Cloud, arc.Options.External)
+	if err != nil {
+		return err
+	}
+
+	// Cloud config
+	cloudConfig, warn, err := cloudapi.GetConsolidatedConfig(
+		test.derivedConfig.Collectors["cloud"], c.gs.Env, "", arc.Options.Cloud, arc.Options.External)
+	if err != nil {
+		return err
+	}
+	if !cloudConfig.Token.Valid {
+		return errors.New( //nolint:golint
+			"not logged in, please login to the Grafana k6 Cloud " +
+				"using the `k6 cloud login` command",
+		)
+	}
+
+	// Display config warning if needed
+	if warn != "" {
+		modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Warning: "+warn))
+	}
+
+	if cloudConfig.Token.Valid {
+		tmpCloudConfig["token"] = cloudConfig.Token
+	}
+	if cloudConfig.Name.Valid {
+		tmpCloudConfig["name"] = cloudConfig.Name
+	}
+	if cloudConfig.ProjectID.Valid {
+		tmpCloudConfig["projectID"] = cloudConfig.ProjectID
+	}
+
+	if arc.Options.External == nil {
+		arc.Options.External = make(map[string]json.RawMessage)
+	}
+
+	b, err := json.Marshal(tmpCloudConfig)
+	if err != nil {
+		return err
+	}
+
+	arc.Options.Cloud = b
+	arc.Options.External[cloudapi.LegacyCloudConfigKey] = b
+
+	name := cloudConfig.Name.String
+	if !cloudConfig.Name.Valid || cloudConfig.Name.String == "" {
+		name = filepath.Base(test.sourceRootPath)
+	}
+
+	logger := c.gs.Logger
+
+	// Start cloud test upload
+	modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Validating script options"))
+	client := cloudapi.NewClient(
+		logger, cloudConfig.Token.String, cloudConfig.Host.String, consts.Version, cloudConfig.Timeout.TimeDuration())
+	if err = client.ValidateOptions(arc.Options); err != nil {
+		return err
+	}
+
+	modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Uploading archive"))
+
+	var cloudTestRun *cloudapi.CreateTestRunResponse
+	cloudTestRun, err = client.UploadTestOnly(name, cloudConfig.ProjectID.Int64, arc)
+	if err != nil {
+		return err
+	}
+
+	refID := cloudTestRun.ReferenceID
```

Comment:
I guess this is out of the scope of this pull request, but do we really want the `k6 cloud upload` command to create a new test run? Or just a test (if not exists)? Is it technically/logically viable?